python/cudf/cudf/core/index.py

```diff
@@ -1808,6 +1971,40 @@ def categories(self):
         """
         return self._values.cat().categories
 
+    @annotate("CATEGORICAL_INDEX_EQUALS", color="green", domain="cudf_python")
+    def equals(self, other):
+        """
+        Determine if two Index objects contain the same elements.
+
+        Returns
+        -------
+        out: bool
+            True if “other” is an Index and it has the same elements
+            as calling index; False otherwise.
+        """
+        if self is other:
+            return True
+        elif not isinstance(other, Index):
+            return False
+        elif len(self) != len(other):
+            return False
+        elif not isinstance(other, Index):
+            return False
```

Comment:
Should we refactor these checks into a utility function since they're used across all of the index `equals` code?